3.9.12 build failure with GCC 14

Trying to build displaycal-py3 3.9.12 with GCC 14 results in the following build error:

```
2024-06-06 16:34:28,105 root INFO x86_64-pc-linux-gnu-cc -Wsign-compare -DNDEBUG -fwrapv -march=native -O2 -pipe -march=native -O2 -pipe -fPIC -DUNIX -I/usr/x86_64-pc-linux-gnu/include/python3.11 -c DisplayCAL/RealDisplaySizeMM.c -o build/temp.linux-x86_64-cpython-311/DisplayCAL/RealDisplaySizeMM.o
DisplayCAL/RealDisplaySizeMM.c: In function 'get_displays':
DisplayCAL/RealDisplaySizeMM.c:871:71: error: passing argument 12 of 'XRRGetOutputProperty' from incompatible pointer type [-Wincompatible-pointer-types]
  871 |                                     &ret_type, &ret_format, &ret_len, &ret_togo, &atomv) == Success
      |                                                                       ^~~~~~~~~
      |                                                                       |
      |                                                                       long unsigned int **
In file included from DisplayCAL/RealDisplaySizeMM.c:33:
/usr/x86_64-pc-linux-gnu/include/X11/extensions/Xrandr.h:339:61: note: expected 'long unsigned int *' but argument is of type 'long unsigned int **'
  339 |                       unsigned long *nitems, unsigned long *bytes_after,
      |                                              ~~~~~~~~~~~~~~~^~~~~~~~~~~
error: command '/usr/bin/x86_64-pc-linux-gnu-cc' failed with exit code 1
```

Switching back to GCC 13 the build works as intended. Please see https://gcc.gnu.org/gcc-14/porting_to.html for details.

--- a/DisplayCAL/RealDisplaySizeMM.c
+++ b/DisplayCAL/RealDisplaySizeMM.c
@@ -848,7 +848,7 @@ disppath **get_displays() {
                         Atom edid_atom, ret_type;
                         int ret_format;
                         long ret_len = 0;
-                        unsigned long *ret_togo;
+                        unsigned long ret_togo;
                         unsigned char *atomv = NULL;
                         int ii;
                         char *keys[] = {        /* Possible keys that may be used */

